openfe/setup/ligand_network_planning.py

```diff
@@ -74,6 +74,26 @@ def generate_radial_network(ligands: Iterable[SmallMoleculeComponent],
     return Network(edges)
 
 
+def maximal_network(
+    ligands: Iterable[SmallMoleculeComponent],
+    mappers: Iterable[LigandAtomMapper],
+    scorer: Optional[Callable[[LigandAtomMapper], float]] = None,
+    # allow_disconnected=True
+):
+    nodes = list(ligands)
+
+    # First create a network with all the proposed mappings (scored)
+    mapping_generator = itertools.chain.from_iterable(
+        mapper.suggest_mappings(molA, molB)
+        for molA, molB in itertools.combinations(nodes, 2)
+        for mapper in mappers
```

Comment:
will this generate multiple edges between two nodes for multiple mappers?

Reply:
Yes. In fact, this is already tested (by way of the minimum spanning graph, which uses this in its first stage). This will try to create all the edges possible, with all the mappers that are provided.

Reply:
Hmm, I would have expected it to instead choose the edge with the best score amongst the mappers... but I can also see this way round too.

Reply:
@RiesBen / @IAlibay would be interested what you'd expect from this and/or is there value in producing a network with multiple edges from different mappers (if so maybe these two different behaviours are controlled by a kwarg)

Reply:
So the only case I personally can think of where this might be useful, is when you end up trying to benchmark various mapping methods against each other. That being said, without any kind of annotation of what the mapper was, it ends up being rather useless.

It's unclear to me what is the intent of this method. Is it:

1. Create a network that just has all the possible connections that could ever be generated (which this does, but will need better annotations).
2. Create as fully connected a network as possible (which this doesn't but could potentially use this method to then filter by score)
3. Something else

If the aim is the first one, then we will simply need a second method that just filters through the results of this method to generate a connected network with a single edge per node pair.

Reply:
ok we're going with multiple and it's documented so all good